lib/tutorials/en_US/testing.md

```diff
@@ -0,0 +1,120 @@
+## Testing
+
+_This tutorial is compatible with hapi v17_
+
+### Testing routes
+
+Hapi is designed for creating robust, testable applications. To this end, Hapi includes the ability to test routes without having to actually start a server, completely avoiding the time overheads and added complexity of the TCP protocol.
+
+This tutorial goes into a basic setup for testing routes, and outlines one possible setup for a testable application.
+
+### The server
+
+Taking the server example from the [Getting Started](https://hapijs.com/tutorials/getting-started) tutorial, we make a minor modification to it, such that it doesn't automatically start when referenced from our tests. We also add a simple route from the [Routing Tutorial](https://hapijs.com/tutorials/routing) that we want to test.
+
+You might call this file `server.js` and place it in the `lib` directory of your project.
+
+```javascript
+'use strict';
+
+const Hapi = require('hapi');
+
+const server = Hapi.server({
+    port: 3000,
+    host: 'localhost'
+});
+
+server.route({
+  method: 'GET',
+  path: '/some/route',
+  handler: function () {
+
+      return 'Hello!';
+  }
+});
+
+exports.init = async () => {
+
+    await server.initialize();
+    return server;
+};
+
+exports.start = async () => {
+
+    await server.start();
+    console.log(`Server running at: ${server.info.uri}`);
+    return server;
+};
+
+process.on('unhandledRejection', (err) => {
+
+    console.log(err);
+    process.exit(1);
+});
+
+```
+
+Note that we call `server.initialize` in our `init` method. We stil want Hapi to set up all our server concerns, such as [caching](https://hapijs.com/tutorials/caching).
+
+Next, we create our main entrypoint for the server. This might be the file referenced by the `main` attribute of your `package.json`. It is run when your application starts. 
+
+```javascript
+'use strict';
+
+const { start, stop } = require('lib/server');
+
+start();
+```
+
+What we've created here is a way of starting the server normally by calling its `start` function in our entrypoint, and exposing a port for external HTTP traffic, but we've also got a module which doesn't do anything by default, which we can use in our tests.
+
+### Writing a route test
+
+In this example we'll use `mocha`, but the same method can be used for any testing tool.
```

Comment:
I would rather make this example use [Lab](https://github.com/hapijs/lab) as it is our test runner from hapi's ecosystem (and [Code](https://github.com/hapijs/code) for the assertion library), but you can also keep the mention about using any other test runner (Jest, Mocha, Tap, ...).

Reply:
Last time I used Lab it wasn't recommended for general use so I didn't know that had changed. I've switched it now. I was already using `code` - I'm not a fan of `chai`.